### PR TITLE
Block PMs to muted target. Fixes #750

### LINKED ISSUE #750: Is PM reciever muted
## Issue
When PlayerA sends a PM to PlayerB, while PlayerB is muted PlayerA does not get any information about this.

## Suggested behaviour

1. PlayerA is not an admin.
2. PlayerA gets a message "PlayerB is muted for another X minutes".
3. PlayerB does not recieve any PM.

### PR DESCRIPTION
**Pull request by JTrolin**
Thank you for making a contribution to [Las Venturas Playground](https://sa-mp.nl/)!

Please take a moment to fill in the questions in this template, to make it easier for our developers
to understand what you're doing.

## What is being changed?
Players sending PMs to a muted target will not be able to, and will be informed that the target is muted and for how long.

## Why is this being changed?
I believe this is an improvement in both terms of transparancy, plus informing the sender that their target is muted and cannot respond.

## How are you making the change?
  [X] I have a working check-out of Las Venturas Playground.
  [X] I have included tests for any changed JavaScript code.
  [X] I have tested this change myself on my local server.

### CODE REVIEW

File: javascript/features/communication_commands/direct_communication_commands.js [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 import { CommandBuilder } from 'components/command_manager/command_builder.js';
 import { ScopedCallbacks } from 'base/scoped_callbacks.js';
+import { MuteManager } from 'features/communication/mute_manager.js';
+import { relativeTime } from 'base/time.js';
```
Comment (RussellLVP) [inline javascript/features/communication_commands/direct_communication_commands.js:8]: nit: please sort these alphabetically. We generally split up classes and functions in separate sections too:

```javascript
import { ClassName } from 'foo/class_name.js';
import { SecondClass } from 'foo/second_class.js';

import { functionName } from 'base/function.js';
```

File: javascript/features/communication_commands/direct_communication_commands.js [hunk 2]
```diff
@@ -20,6 +22,8 @@ export class DirectCommunicationCommands {
     communication_ = null;
     nuwani_ = null;
     playground_ = null;
+    
+    muteManager_ = new MuteManager();
```
Comment (RussellLVP) [inline javascript/features/communication_commands/direct_communication_commands.js]: nit: delete, also enables you to delete the `import { MuteManager }` line

File: javascript/features/communication_commands/direct_communication_commands.test.js [hunk 3]
```diff
@@ -158,6 +158,18 @@ describe('DirectCommunicationCommands', (it, beforeEach) => {
...
+    it('cannot send to muted targets unless sender is an admin', async (assert) => {
+        const playground = server.featureManager.loadFeature('playground');
+        
+        // Set up by Russell muting themselves
+        assert.isTrue(await russell.issueCommand('/mute Russell 5'));
+
+        // (1) It should not be possible for Gunther to PM Russell who's muted
+        assert.isTrue(await gunther.issueCommand('/pm Russell Hello world'));
+        assert.equal(gunther.messages.length, 1);
+        assert.equal(gunther.messages[0],Message.format(Message.COMMUNICATION_PM_TARGET_MUTED, russell.name, russell.id, "5 minutes"));
```
Comment (RussellLVP) [inline javascript/features/communication_commands/direct_communication_commands.test.js]: nit: we have a line length limit of 100 characters, and please use single quotes.

```javascript
    assert.equal(
        gunther.messages[0],
        Message.format(Message.COMMUNICATION_PM_TARGET_MUTED, russell.name, russell.id,
                       '5 minutes'));
```

**Review by RussellLVP — APPROVED** [review]
Ace, thank you! A few small comments

File: javascript/features/communication_commands/direct_communication_commands.js (hunk 2 above)
Comment (JTrolin) [reply javascript/features/communication_commands/direct_communication_commands.js]: Fixed as of b10fd3ed14f79bc2e46618c53df2363d2a7ba280

File: javascript/features/communication_commands/direct_communication_commands.js (hunk 1 above)
Comment (JTrolin) [reply javascript/features/communication_commands/direct_communication_commands.js:8]: Fixed as of b10fd3ed14f79bc2e46618c53df2363d2a7ba280

File: javascript/features/communication_commands/direct_communication_commands.test.js (hunk 3 above)
Comment (JTrolin) [reply javascript/features/communication_commands/direct_communication_commands.test.js]: Fixed as of 8730b90088d43baa26d1f881561c7d54a913bdf6

**Review by RussellLVP — APPROVED** [review]
Thank you!

**Pull request merged**